agent/lm_agent/backend_utils.py

```diff
@@ -12,6 +13,20 @@
 from lm_agent.logs import logger
 
 
+def _get_cache_dir() -> pathlib.Path:
+    """
+    Get the path to the cache dir. Will always be a sub-directory of the current user's home.
+    """
+    return pathlib.Path.home() / f".cache/license-manager"
```

Comment:
LM user is a system user, not a normal one. `$HOME` might  not be defined and we might not have permissions to create this folder, specially in a shared environment with network mounted `/home`.

Why changing from `/run`?

Reply:
Addressed this issue in the latest commit. CACHE_DIR restored as an environment variable. No logic to create missing directories. If the cache dir doesn't exist, you don't get cached tokens.